standard/interfaces.md

```diff
@@ -546,41 +546,21 @@ For a type `T` that is a struct or a class that implements interfaces `I2` and `
 >     void IA.M() { Console.WriteLine("IC.M"); }
 > }
 > 
-> abstract class C: IB, IC { } // error: no most specific override for 'IA.M'
+> abstract class C: IB, IC { } // error: no most specific implementation for 'IA.M'
 > 
 > abstract class D: IA, IB, IC // OK
 > {
 >     public abstract void M();
 > }
 > ```
 >
-> The most specific override rule ensures that a conflict (i.e., an ambiguity arising from diamond inheritance) is resolved explicitly by the programmer at the point where the conflict arises. *end example*
-
-It is an error if in a class declaration the most specific override of some interface method is an abstract override that was declared in a class or an interface.
-
-> *Example*:
->
-> <!-- Example: {template:"standalone-lib-without-using", name:"InterfaceMethods5", expectedErrors:["CS0535"]} -->
-> ```csharp
-> interface IF
-> {
->     void M();
-> }
-> 
-> abstract class F : IF { } // error: 'F' does not implement 'IF.M'
-> ```
->
-> *end example*
+> The most specific implementation rule ensures that a conflict (i.e., an ambiguity arising from diamond inheritance) is resolved explicitly by the programmer at the point where the conflict arises. *end example*
 
 ### 19.4.6 Interface member access
 
 Interface members are accessed through member access ([§12.8.7](expressions.md#1287-member-access)) and indexer access ([§12.8.12.4](expressions.md#128124-indexer-access)) expressions of the form `I.M` and `I[A]`, where `I` is an interface type, `M` is a constant, field, method, property, or event of that interface type, and `A` is an indexer argument list.
 
-In a class `D`, with direct or indirect base class `B`, where `B` directly or indirectly implements interface `I` and `I` defines a method `M()`:
-
-- The expression `base.M()` is valid only if `B` provides a definition for `M`. The expression `base.M()` is valid only if the most specific implementation (§most-specific-implementation) of `M` is in a class type. To access the most specific implementation if it was defined in an interface, use the cast expression `(this as I).M()` to refer to the most specific implementation (§most-specific-implementation) of `M` in `I` or an interface derived from `I`.
-- If `B` contains a method `M` without the `virtual` or `override` modifier, that method hides `I.M()` rather than overriding it.
-- If `B` contains a method `M` with a different parameter list, that method `M` is a distinct overload of `M`.
+In a class `D`, with direct or indirect base class `B`, where `B` directly or indirectly implements interface `I` and `I` defines a method `M()`, the expression `base.M()` is valid only if the most specific implementation (§most-specific-implementation) of `M()` is in a class type.
```

Comment:
It's much clearer to say that the validity of `base.M()` depends entirely on whether some base class defines _its own_ member `M()`. Whether or not its own member `M()` is an implementation of some `I.M()` does not affect the validity of `base.M()` one way or another.

Reply:
Also, Here's an example where the expression base.M() is valid, but the most specific implementation of `I.M()` is not in a class type:
```cs
interface I 
{ 
    void M() => Console.WriteLine("impl");
}

class B : I
{
    public Action M => () => Console.WriteLine("Unrelated impl via a property");
}

class D : B
{
    void Example() => base.M(); // This is not I.M();
}
```